### DISCUSSION TITLE
Disabling key repeat

### DISCUSSION THREAD

**Discussion by burbma** [Textualize/textual#3771] [Q&A] [answered]
Any ideas on how to disable key repeats in my input? It seems `Input`'s `on_key` can't tell the difference between me repeatedly pressing a key and holding it down to trigger key repeat. I assume the OS is sending the same events either way. I would like to be able to handle this in Textual rather than via the OS so some apps can utilize the user holding a key down to type it many times while other apps only type the key once no matter how long it's held down.

**Comment by TomJGooding** [comment, marked as answer]
I don't think it is possible to distinguish repeatedly pressing a key and holding it down. This is because of limitations in the terminal rather than Textual, as discussed here:

- #3125
